crates/verifier/src/groth16/mod.rs

```diff
@@ -69,11 +70,23 @@ impl Groth16Verifier {
 
         let sp1_vkey_hash = decode_sp1_vkey_hash(sp1_vkey_hash)?;
 
-        Self::verify_gnark_proof(
+        // It is computationally infeasible to find two distinct inputs, one processed with
+        // SHA256 and the other with Blake3, that yield the same hash value.
```

Comment:
```suggestion
        // First, check if the public values hashed with SHA2 match the expected public values. If not, try hashing 
       // with Blake3. If both fail, return an error. We perform the checks sequentially to avoid calculating both 
       // hashes unless necessary.
```